core/src/main/java/com/github/mizool/core/exception/UnprocessableEntityException.java

```diff
@@ -1,9 +1,13 @@
 package com.github.mizool.core.exception;
 
 /**
- * Thrown when the entity is syntactically correct (e.g. all non-null properties are set), but could not be processed
- * for semantic reasons (e.g. invalid references to other database entities).
+ * @deprecated It is designed to be thrown when the entity is syntactically correct (e.g. all non-null properties are set),
+ * but could not be processed for semantic reasons (e.g. invalid references to other database entities).
+ * But the UnprocessableEntityException doesn't really say why the entity is unprocessable and
+ * is generally a bad fit as no client ever really has a chance of handling such a response. Try to avoid using this
+ * exception and use more specific one.
  */
+@Deprecated
```

Comment:
Although it's the default value, add `(forRemoval = false)` to indicate that we consciously decided to keep the class but simply discourage its use.

Reply:
`(forRemoval = false)` is available only from Java 9

Reply:
It's like being in the dark ages....

Reply:
When discussing this, my idea was that the more specific exceptions that applications define should extend this class. This way, one would not have to define exception behaviors each time one creates such an exception.

Deprecating the class would create compiler warnings in all those concrete exceptions due to `extends UnprocessableEntityException`. We could then of course add a `@SuppressWarnings("deprecation")`, but that feels quite weird because we don't want to discourage _extending the class_, but discourage _creating instances of it_.

My next thought was to deprecate the constructors instead of the class. However, that still doesn't quite work because then the `super(...)` calls in the constructors of subclasses would trigger compiler warnings.

So it's not easy. Still, I very much would like to have one base class to make it really convenient to create specific exceptions.

So here's my next idea: How about we keep the deprecation of this class, but introduce a new class `AbstractUnprocessableEntityException`?

- The existing exception behavior would be changed to use  `AbstractUnprocessableEntityException` instead.
- Both the deprecated not-specific-enough `UnprocessableEntityException` and any properly-specific application-provided `FooException` extend  `AbstractUnprocessableEntityException`.

Reply:
I think this is good way to go. That way we
- will avoid extending UnprocessableEntityException which would cause all compiler warnings
- support extending and discourage creating instances of UnprocessableEntityException
- will be able to completely get rid of UnprocessableEntityException in the future

Reply:
I like.